analysis/quantifier-comparisons/09-emptydrop-comparisons.Rmd

```diff
@@ -0,0 +1,461 @@
+Then we will look at the distribution of UMI/cell and genes/cell with different filtering strategies in comparison to Cell Ranger. 
+
+### Number of Cells Filtered
+
+```{r}
+# function to grab the number of cells from an sce object 
+get_num_cells <- function(sce){
+  num_cells <- dim(sce)[2]
+}
+```
+
+
+```{r}
+emptydrops_cell_num <- emptydrops_filter_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(emptydrops_cell_num) <- c("emptyDrops")
+
+emptydrops_200_cell_num <- emptydrops_filter_sces_200 %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(emptydrops_200_cell_num) <- c("emptyDrops200")
+
+emptydrops_500_cell_num <- emptydrops_filter_sces_500 %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(emptydrops_500_cell_num) <- c("emptyDrops500")
+
+cr_filter_cell_num <- crlike_filter_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(cr_filter_cell_num) <- c("emptyDropsCellRanger")
+
+knee_filter_cell_num <- knee_filtered_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame() 
+rownames(knee_filter_cell_num) <- c("knee")
+
+names(cellranger_sces) <- sample_names[3:6]
+# add additional nuclei samples from previous benchmarking 
+cellranger_sces[["SCPCR000118-nuclei"]] <- cellranger_nuclei_sces[["SCPCR000118-cdnapre_mRNA"]]
+cellranger_sces[["SCPCR000119-nuclei"]] <- cellranger_nuclei_sces[["SCPCR000119-cdnapre_mRNA"]]
+                                                                  
+cellranger_cell_num <- cellranger_sces %>%
+  purrr::map(get_num_cells) %>%
+  as.data.frame()
+rownames(cellranger_cell_num) <- c("CellRanger")
+
+# create combined dataframe with number of cells and filtering method 
+combined_cell_num_df <- dplyr::bind_rows(emptydrops_cell_num,
+                                         emptydrops_200_cell_num,
+                                         cr_filter_cell_num, 
+                                         knee_filter_cell_num,
+                                         cellranger_cell_num,
+                                         emptydrops_500_cell_num) %>%
+  tibble::rownames_to_column("filtering_method") %>%
+  tidyr::pivot_longer(cols = starts_with("SCPCR"),
+                      names_to = "sample", 
+                      values_to = "number_cells")
+```
+
+
+```{r fig.height=7, fig.width=10}
+ggplot(combined_cell_num_df, aes(x = sample, y = number_cells, color = filtering_method)) + 
+  geom_jitter() +
+  geom_sina(size = 4) + 
+  theme(axis.text.x = element_text(angle = 90))
+```
+Here you can see that `emptyDropsCellRanger()` actually does appear to give more comprable cell numbers to Cell Ranger than `emptyDrops()`.
+Additionally, as expected, using the knee-based method decreases the numbers of cells across the board. 
+It appears that `emptyDrops()` with `lower=200` or `lower=500` is the most comparable to Cell Ranger for all samples and we see that those cell numbers tend to be fairly similar. 
```

Comment:
Except for #118, where `emptyDrops` does do much "worse" than `emptyDropsCellRanger` in that it calls many fewer cells. I'm not sure if there are other emptyDrops parameters we might be able to adjust to deal with pathological cases like that? I just looked through and can't really find anything. This does suggest that we might actually want to think a bit more about emptyDropsCellRanger, but it will depend how fast it gets into being something other people can use. 

I'm not actually convinced retaining those low UMI count cells is worthwhile though, so I might not be too upset with the "stricter" emptydrops filtering. We can be a bit more lenient with lower = 200, but nothing is necessarily going to save a sample like 118...

Reply:
I would agree that retaining the low UMI count cells is not worthwhile and I'm inclined to go with `lower=500` since those cells should be filtered out anyways. I also don't think anything would save a sample like 118 here either and that's a case where it is just a poor sample, so I think we should design to the standard sample that we expect to get.